packages/integration-sdk-cli/src/commands/diff.ts

```diff
@@ -184,6 +258,16 @@ function diffRelationships(
 
   const newRelationships: DiffableRelationships = {};
   for (const edge of newEdges) {
+    if (options?.ignoreSystemMapper) {
```

Comment:
It appears that very similar logic is being repeated. Can we extract some of this out and include unit tests?